### ISSUE TITLE
False positive when comparing ReflectionNamedType::getName() with 'static'

### ISSUE THREAD

**Issue by BenMorel** [vimeo/psalm#8167]
When `ReflectionNamedType::isBuiltin()` returns `false`, Psalm assumes that `getName()` returns a `class-string`:

```php
if ($returnType instanceof \ReflectionNamedType) {
    if (! $returnType->isBuiltin() && $returnType->getName() !== 'static') {
        
    }
}
```

> Psalm output (using commit 8b7bc07): 
> 
> ERROR: [RedundantCondition](https://psalm.dev/122) - 16:39 - 'static' can never contain class-string

Demo: https://psalm.dev/r/2c06d6b1d3

Even though `'static'` is also a valid return type since PHP 8.0: https://3v4l.org/ZAqYK

**Comment by AndrolGenhald** [comment]
Looks to me like the assertion in the stub needs changed from `class-string` to `class-string|'static'|'self'`. Are there any other possible return values we're missing?

**Comment by BenMorel** [comment]
@AndrolGenhald Good point about `self`, I thought this would be replaced with the FQCN by reflection or even at compilation time, but it looks like it's not, it's actually returning `'self'`: https://3v4l.org/U3JtZ

I don't see any other possible return value, so `class-string|'static'|'self'` should be alright.

I couldn't find where the return type is documented, it's not in the [CallMap](https://raw.githubusercontent.com/vimeo/psalm/4.x/dictionaries/CallMap.php) at least. Because the result is dependent on whether `isBuiltin()` has been called first, I guess there's some custom logic around this class.

**Comment by AndrolGenhald** [comment]
It's in `stubs/Reflection.phpstub`, it just uses `@psalm-assert-if-false` to assert the type of `$this->getName()`.

**Comment by BenMorel** [comment]
@AndrolGenhald TIL, thanks! Fixed in #8201.
